pkg/synthesis/synthesis_test.go

```diff
@@ -102,208 +120,249 @@ var groupsByVmsTests = []synthesisTest{
+}
+
+// parallelTestsRun() gets a test function to run, and run it on all the syntheticTests in parallel
+func parallelTestsRun(t *testing.T, f func(synTest *synthesisTest, t *testing.T, rc *collector.ResourcesContainerModel)) {
+	logging.Init(logging.HighVerbosity)
+	for _, test := range allSyntheticTests {
+		rc := data.ExamplesGeneration(&test.exData.FromNSX)
+		t.Run(test.name, func(t *testing.T) {
+			f(&test, t, rc)
+		},
+		)
 	}
-	baseName := fmt.Sprintf("%s_%t_%d", synTest.name, synTest.noHint, synTest.allowOnlyFromCategory)
-	outDir := path.Join("out", baseName)
-	fmt.Println("suffix:", suffix)
-	abstractModel, err := NSXToK8sSynthesis(rc, outDir, hintsParm, synTest.allowOnlyFromCategory)
-	expectedOutputFileName := filepath.Join(getTestsDirOut(), synTest.name+suffix)
-	expectedOutputDir := filepath.Join(getTestsDirOut(), k8sResourcesDir, baseName)
-	require.Nil(t, err)
-	addDebugFiles(t, rc, abstractModel, outDir)
-	actualOutput := strAllowOnlyPolicy(abstractModel.policy[0])
-	fmt.Println(actualOutput)
-	compareOrRegenerateOutputPerTest(t, mode, actualOutput, expectedOutputFileName, synTest.name)
-	compareOrRegenerateOutputDirPerTest(t, mode, filepath.Join(outDir, k8sResourcesDir), expectedOutputDir, synTest.name)
 }
 
-func addDebugFiles(t *testing.T, rc *collector.ResourcesContainerModel, abstractModel *AbstractModelSyn, outDir string) {
-	connectivity := map[string]string{}
-	var err error
-	// generate connectivity analysis output from the original NSX resources
-	debugDir := path.Join(outDir, "debug_resources")
+//////////////////////////////////////////
+// the test functions:
+//////////////////////////////////////////
 
-	for _, format := range []string{"txt", "dot"} {
-		params := common.OutputParameters{
-			Format: format,
-		}
-		connectivity[format], err = model.NSXConnectivityFromResourcesContainer(rc, params)
-		require.Nil(t, err)
-		err = common.WriteToFile(path.Join(debugDir, "vmware_connectivity."+format), connectivity[format])
-		require.Nil(t, err)
+func runPreprocessing(synTest *synthesisTest, t *testing.T, rc *collector.ResourcesContainerModel) {
+	// get the config:
+	parser := model.NewNSXConfigParserFromResourcesContainer(rc)
+	err := parser.RunParser()
+	require.Nil(t, err)
+	config := parser.GetConfig()
+	// write the config summery into a file, for debugging:
+	configStr := config.GetConfigInfoStr(false)
+	err = common.WriteToFile(path.Join(synTest.debugDir(), "config.txt"), configStr)
+	require.Nil(t, err)
+	// get the preProcess results:
+	categoryToPolicy := preProcessing(config.Fw.CategoriesSpecs)
+	preProcessOutput := stringCategoryToSymbolicPolicy(config.Fw.CategoriesSpecs, categoryToPolicy)
+	fmt.Println(preProcessOutput)
+	// write the preProcess results into a file, for debugging:
+	err = common.WriteToFile(path.Join(synTest.debugDir(), "pre_process.txt"), preProcessOutput)
+	require.Nil(t, err)
+	// compare to expected results:
+	if synTest.hasExpectedResults() {
```

Comment:
if there are no expected results, no need to generate them?

Reply:
currently there are two tests with hasExpectedResults:
1. reading from a resources.json (which is currently empty)
2. reading from the server